src/controllers/__tests__/item.test.js

```diff
@@ -365,19 +408,22 @@ describe('Items Endpoints', () => {
     test('Should response with an error message when to be updated item ID not found', async () => {
       const response = await request(server)
         .put('/api/items/6208e47a5fe21cc475419234')
+        .set('Cookie', authCookie)
         .send({ isAvailable: false });
 
       expect(response.header['content-type']).toContain('application/json');
-      expect(response.statusCode).toBe(422);
+      expect(response.statusCode).toBe(401);
       expect(response.body.message).toBe(
-        'The item with the specified ID was not found.'
+        'unauthorized to modify requested item'
```

Comment:
This got me thinking...
Wasn't the previous error message more well-defined? It tells you exactly what the issue is.
The middleware error message tells the user they're unauthorized but doesn't mention the reason.
This could be confusing for frontend users when they integrate with your API.

Can the error messages in the middleware be updated to say something like `unauthorized to modify requested item: item not found` and similar for the other cases.

Reply:
Ok I can change it. but sometimes there will be an item with the specified Id but the user is not the real owner, is it ok to send `item not found` for both cases?

Reply:
For that case, the reason can say `only item owner can modify`.